core/templates/App.ts

```diff
@@ -89,32 +89,67 @@ angular.module('oppia').config([
     // used in the directives imported in this file and cannot be
     // injected before bootstrapping of oppia module.
     var servicesToProvide = [
-      'AlertsService', 'BackgroundMaskService', 'BrowserCheckerService',
-      'CodeReplRulesService', 'CollectionCreationBackendService',
-      'CollectionCreationBackendService',
-      'ContextService', 'CreatorDashboardBackendApiService', 'CsrfTokenService',
-      'DateTimeFormatService', 'DebouncerService', 'DeviceInfoService',
-      'DocumentAttributeCustomizationService',
+      'AlertsService', 'AngularNameService',
```

Comment:
There are a lot of additions to this array.  Any reason why?

Reply:
When changing controllers to AngularJS components, I need to list all the dependencies, the dependencies of the dependencies and so on in this array otherwise it'll throw an error about Angular injector.

Reply:
Alright! In the array, there is `TopicViewerBackendApiService`. If I am not wrong this is only used in the topic-viewer-page module. However, `App.ts` is included in every module script. So basically this would be present in all modules even when not required. This could be done by creating another `servicesToProvide` array in `topic-view-page.scripts.ts` right after `App.ts` is required or at the very end in `topic-viewer-page.module.ts`. Personally, I am a bit skeptical about the second method. I don't think this would create much of a difference but just a suggestion.
+cc @bansalnitish

Reply:
That's nice! However, if you think it's not going to create much of a difference, I think it's ok to keep all the services in `App.ts`